feat(curriculum): add password generator project to Python curriculum (#52444)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/20-upcoming-python/learn-regular-expressions-by-building-a-password-generator/6577319039f4f7de9251b822.md

---
id: 6577319039f4f7de9251b822
title: Step 61
challengeType: 20
dashedName: step-61
---

# --description--

Instead of using a loop and a counter variable, you can achieve the same result with a different approach.

Replace your existing `for` loop and two `if` statements with a single `if` statement. For the `if` condition, use a list created with the list comprehension syntax.

> The list should be made of the results of evaluating the expression `constraint <= len(re.findall(pattern, password))` for each `constraint`-`pattern` couple in the `constraints` list.

Reviewer: ```suggestion
The list store the results of evaluating the expression `constraint <= len(re.findall(pattern, password))` for each `constraint`-`pattern` tuple in the `constraints` list.
```

# --hints--

You should replace your existing `for` loop and two `if` statements with a single `if` statement. 

```js
({ test: () => {
  assert.match(e.code.original["main.py"], /^(\s{8})if\s+.+:\s*^\1\s{4}break/ms);
  assert.isFalse(/if\s+count\s*==\s*4\s*:/.test(code));
} })
```

> Your new `if` condition should be a list comprehension that is the result of evaluating the expression `constraint <= len(re.findall(pattern, password))` for each `constraint`, `pattern` in the `constraints` list.

Reviewer: ```suggestion
Your new `if` condition should be a list comprehension that is the result of evaluating the expression `constraint <= len(re.findall(pattern, password))` for each `constraint`-`pattern` tuple in the `constraints` list.
```
